instrumentation/github.com/aws/aws-sdk-go-v2/otelaws/snsattributes_test.go

```diff
@@ -0,0 +1,74 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package otelaws
+
+import (
+	"context"
+	"testing"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/sns"
+	"github.com/aws/smithy-go/middleware"
+	"github.com/stretchr/testify/assert"
+
+	semconv "go.opentelemetry.io/otel/semconv/v1.17.0"
+)
+
+func TestSNSPublishInputTargetARN(t *testing.T) {
```

Comment:
All those tests are extremely similar. How about DRYing them up with subtests?

Reply:
valid point, thank you, I simply followed the tests written for SQS so it was not dry